rain_api_core/egress_util.py

```diff
@@ -163,7 +163,7 @@ def check_public_bucket(bucket, b_map, optional_uri=""):
         log.debug('we have a PUBLIC_BUCKETS in the ordinary bucketmap file')
         for pub_bucket in b_map['PUBLIC_BUCKETS']:
             # log.debug('is {} the same as {}?'.format(bucket, prepend_bucketname(pub_bucket)))
-            if bucket_prefix_ismatch(bucket, pub_bucket, optional_uri):
+            if bucket_prefix_ismatch(bucket, pub_bucket, optional_uri) and bucket == prepend_bucketname(pub_bucket):
```

Comment:
Will this return `True` if `BUCKETNAME_PREFIX='prod'` and the bucket_map is: 
```
MAP:
    product: productbucket
PUBLIC_BUCKET:
   productbucket/browse: "this is public data"
```
...and a request comes in for
`https://<tea>/product/browse/image/granule.png`?

What are the for `bucket`, `pub_bucket`, `optional_uri`, and the return value from `prepend_bucketname(pub_bucket)`?